dcos/marathon.py

```diff
@@ -23,6 +23,10 @@ def create_client(config=None):
 
     marathon_url = _get_marathon_url(config)
 
+    if 'marathon.username' in config:
+        http.auth = (config.get('marathon.username'),
```

Comment:
This looks like it will affect anything else that uses HTTP - which most likely won't need authentication set. I.e., If I use Marathon auth, and run a Marathon command, then later try to do a `dcos package update`, http.auth will be set. Would it be better to pass in the auth object when we make HTTP calls so it doesn't stick around?